libs/wingsdk/src/shared-aws/domain.ts

```diff
@@ -17,15 +17,48 @@ export class Domain extends cloud.Domain {
   constructor(scope: Construct, id: string, props: cloud.DomainProps) {
     super(scope, id, props);
 
-    const values = getPlatformSpecificValues(
-      this,
-      "iamCertificate||acmCertificateArn",
-      "hostedZoneId"
+    const registrar = App.of(scope).platformParameterRegistrar;
+
+    // Domain requires parameters from the user, so we need to add the parameter schemas to the registrar
+    let s = {
+      type: "object",
+      oneOf: [
+        {
+          required: ["iamCertificate"],
+        },
+        {
+          required: ["acmCertificateArn"],
+        },
+      ],
+      required: ["hostedZoneId"],
+      properties: {
+        iamCertificate: {
+          type: "string",
+        },
+        acmCertificateArn: {
+          type: "string",
+        },
+        hostedZoneId: {
+          type: "string",
+        },
+      },
+    };
+
+    registrar.addParameterSchemaAtPath(s, this.node.path, true);
+
+    const iamCertificate = registrar.getParameterValue(
+      `${this.node.path}/iamCertificate`
+    );
+    const acmCertificateArn = registrar.getParameterValue(
+      `${this.node.path}/acmCertificateArn`
+    );
+    const hostedZoneId = registrar.getParameterValue(
+      `${this.node.path}/hostedZoneId`
     );
```

Comment:
I wonder if it's possible to do schema validation when the user calls `addParameterSchemaAtPath`? This could prevent bugs where the user puts a value of the wrong kind in their parameters file (like a list instead of a string), and this code and any code that depends on the values would run without errors

Reply:
It could, though from a dx perspective to the wing user I think that would result in them getting partial validation errors. Since right now validation happens all at once in an `allOf` block during preSynth we can return a full back to the user of all their parameter issues, rather than show them partial, have them fix then get the rest of the missing issues.

Reply:
Maybe eventually we can refactor `getParameterValue` to return a token of some kind